src/main/java/org/spongepowered/common/event/manager/SpongeEventManager.java

```diff
@@ -325,8 +325,13 @@ public SpongeEventManager registerListeners(final PluginContainer plugin, final
 
     @Override
     public SpongeEventManager registerListeners(final PluginContainer plugin, final Object listener,
-                                                final MethodHandles.Lookup lookup) {
+                                                MethodHandles.Lookup lookup) {
         Objects.requireNonNull(lookup, "lookup");
+        try {
+            lookup = MethodHandles.privateLookupIn(listener.getClass(), lookup);
+        } catch (IllegalAccessException ex) {
+            throw new IllegalArgumentException("Provider lookup lacks sufficient privilege", ex);
+        }
```

Comment:
I'm a fan of having final variables, I'd suggest not unfinalizing the one here, and providing a new variable for the `usedLookup` instead.

I'd also suggest to include a string message for the plugin's container information to help identify what plugin was causing the issue or supplying the inaccessible lookup.